BlueprintUI/Sources/Element/ComparableElement.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import UIKit
+
+
+/// An `ComparableElement` is an element which can cache its
+/// measurements and layouts across layout passes in order to dramatically improve
+/// performance of repeated layouts.
+///
+/// Some element types within Blueprint already conform to `ComparableElement`.
+/// If your element can be easily compared for equality, consider making your
+/// it conform to `ComparableElement` for improved performance across layouts.
+public protocol ComparableElement: AnyComparableElement {
+
+    ///
+    /// Indicates if the element is equivalent to the other provided element.
+    /// Return `true` if the elements are the same, or return `false` if something about
+    /// the element changed that will affect measurement or layout.
+    ///
+    /// ## Note
+    /// Even if this method returns true, the `ViewDescription`
+    /// backing the element will still be re-applied to the on-screen view.
```

Comment:
This was another thing I remember playing with – allowing to skip application of the `element` to the view if `isEquivalent` is true, which can also be a minor perf bump.